v3_async_wip/v3_async_wip/mqtt_client.py

```diff
@@ -386,6 +390,14 @@ def is_connected(self) -> bool:
         """
         return self._connected
 
+    def previous_disconnection_cause(self) -> Optional[MQTTError]:
+        """
+        Returns an MQTTError from the previous disconnection if it was unexpected as a result of
+            a connection drop.
+        Returns None if the previous disconnection attempt was intentional
```

Comment:
> Returns None if the previous disconnection attempt was intentional [](http://example.com/codeflow?start=7&length=67)

It looks like generators don't exit if the disconnect was intentional. Don't we want them to exit?